### PR TITLE
Update electronic horizon example to new API

### PR DESCRIPTION
**Pull request by azarovalex**
Electronic horizon public API was changed here https://github.com/mapbox/mapbox-navigation-ios/issues/2906.
This PR changes the "Upcoming intersections" example to use the new API.

### CODE REVIEW

File: Navigation-Examples/Examples/Upcoming-Intersection.swift [hunk 1]
```diff
@@ -21,23 +21,14 @@ class ObservingElectronicHorizonEventsViewController: UIViewController {
...
+        navigationMapView.mapView.location.overrideLocationProvider(with: passiveLocationManager)
         navigationMapView.mapView.update {
             $0.location.puckType = .puck2D()
         }
         navigationMapView.mapView.on(.styleLoaded) { [weak self] _ in
             self?.setupMostProbablePathStyle()
         }
         
-        // TODO: Provide a reliable way of setting camera to current coordinate.
-        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) {
```
Comment (azarovalex) [inline Navigation-Examples/Examples/Upcoming-Intersection.swift:33]: In the latest NavSDK version camera API was changed and this code doesn't compile anymore.
This code will be rewritten using new API here: https://github.com/mapbox/mapbox-navigation-ios-examples/issues/101

**Comment by MaximAlien** [comment]
Shall we separate changes related to SwiftLint and Electronic Horizon? (e.g. move changes which are needed for SwiftLint to another PR)? I think camera related changes already landed to `release-v2.0`, so rebasing is needed.

**Comment by S2Ler** [comment]
@azarovalex mind reviving this PR?

**Comment by azarovalex** [comment]
After updating this PR with the latest changes from the main it looks like the main branch already has all the changes, so I'll close the PR.

**Pull request closed without merging**